### PR TITLE
MIRV warhead

### PR DESCRIPTION
**Pull request by 70000hp**
COC: 
Improved that ancient MIRV entity used for the old missile
New MIRV warhead and recipe
The mirvlets also chunkload themselves

### CODE REVIEW

File: src/main/java/com/hbm/inventory/recipes/AssemblerRecipes.java [hunk 1]
```diff
@@ -442,6 +442,23 @@ private static void registerDefaults() {
...
 		makeRecipe(new ComparableStack(ModItems.mp_warhead_15_incendiary, 1), new AStack[] {new ComparableStack(ModItems.seg_15, 1), new OreDictStack(STEEL.plate(), 16), new ComparableStack(ModBlocks.det_charge, 2), new OreDictStack(P_RED.dust(), 8), new ComparableStack(ModItems.circuit_targeting_tier3, 1), },200);
 		makeRecipe(new ComparableStack(ModItems.mp_warhead_15_nuclear, 1), new AStack[] {new ComparableStack(ModItems.seg_15, 1), new OreDictStack(STEEL.plate(), 24), new OreDictStack(TI.plate(), 12), new OreDictStack(PU239.ingot(), 3), new ComparableStack(ModBlocks.det_charge, 6), new ComparableStack(ModItems.circuit_targeting_tier4, 1), },500);
+		makeRecipe(new ComparableStack(ModItems.mp_warhead_15_n2, 1), new AStack[] {new ComparableStack(ModItems.seg_15, 1), new OreDictStack(STEEL.plate(), 8), new OreDictStack(TI.plate(), 20), new ComparableStack(ModBlocks.det_charge, 24), new ComparableStack(Blocks.redstone_block, 12), new OreDictStack(MAGTUNG.dust(), 6), new ComparableStack(ModItems.circuit_targeting_tier4, 1), },400);
```
Comment (HbmMods) [inline src/main/java/com/hbm/inventory/recipes/AssemblerRecipes.java:445]: a duplicate?

File: src/main/java/com/hbm/tileentity/turret/TileEntityTurretBase.java [hunk 2]
```diff
@@ -100,10 +101,10 @@ public void updateEntity() {
 	}
 	
 	private boolean isInSight(Entity e) {
-		if(!(e instanceof EntityLivingBase) && !(e instanceof EntityMissileBaseAdvanced) && !(e instanceof EntityBomber) && !(e instanceof EntityMissileCustom))
+		if(!(e instanceof EntityLivingBase) && !(e instanceof EntityMissileBaseAdvanced) && !(e instanceof EntityBomber) && !(e instanceof EntityMissileCustom)&& !(e instanceof EntityMIRV))
 			return false;
 		
-		if(this instanceof TileEntityTurretCIWS && !(e instanceof EntityMissileBaseAdvanced) && !(e instanceof EntityBomber) && !(e instanceof EntityMissileCustom))
+		if(this instanceof TileEntityTurretCIWS && !(e instanceof EntityMissileBaseAdvanced) && !(e instanceof EntityBomber) && !(e instanceof EntityMissileCustom)&& !(e instanceof EntityMIRV))
```
Comment (HbmMods) [inline src/main/java/com/hbm/tileentity/turret/TileEntityTurretBase.java]: i don't even think this class is even used properly anymore

File: src/main/java/com/hbm/entity/missile/EntityMissileCustom.java [hunk 3]
```diff
@@ -293,7 +296,59 @@ public void onUpdate() {
...
+					return;    
+    			               
+    				this.setDead();
+    				           
+    				double mod;
+    				double mod2;
+    				for(int i = 0; i < 7; i++) {
+    					EntityMIRV nuke3 = new EntityMIRV(this.worldObj);
+    					nuke3.setPosition(posX,posY,posZ);      
+    					mod = (i==(1 & 2)) ? 1 : -1;
```
Comment (HbmMods) [inline src/main/java/com/hbm/entity/missile/EntityMissileCustom.java]: `&` is a bit operator. did you mean `i == 1 || i == 2`?

File: src/main/java/com/hbm/tileentity/turret/TileEntityTurretBase.java (hunk 2 above)
Comment (70000hp) [reply src/main/java/com/hbm/tileentity/turret/TileEntityTurretBase.java]: Weirdly changing that one worked to make it target

File: src/main/java/com/hbm/entity/missile/EntityMissileCustom.java (hunk 3 above)
Comment (70000hp) [reply src/main/java/com/hbm/entity/missile/EntityMissileCustom.java]: Yup, that explains why I had a massive headache trying to make the +X +Z quadrant mirvlet work

File: src/main/java/com/hbm/tileentity/turret/TileEntityTurretBase.java (hunk 2 above)
Comment (70000hp) [reply src/main/java/com/hbm/tileentity/turret/TileEntityTurretBase.java]: oh nvm thats for the phalanx

**Comment by 70000hp** [comment]
Alright, MIRV is ready to go

**Comment by JoeStalini** [comment]
Where is MIRV?

**Pull request closed without merging**